src/main/java/skyler/Parser.java

```diff
@@ -0,0 +1,77 @@
+package skyler;
+
+/**
+ * Represents an interpreter that makes sense of user commands
+ */
+public class Parser {
+
+    private TaskList taskList;
+
+    /**
+     * Creates a parser object
+     *
+     * @param taskList TaskList to be manipulated with user commands.
+     */
+    public Parser(TaskList taskList) {
+        this.taskList = taskList;
+    }
+
+    /**
+     * Returns status number based on input command detected
+     *
+     * @param command User command.
+     * @return 1 if user has more commands and 0 if user has completed commands.
+     * @throws EmptyDescriptionException If no input task is detected after command.
+     * @throws TaskNotRecognisedException If command is not recognised.
+     */
+    public int parse(String command) throws EmptyDescriptionException, TaskNotRecognisedException {
+        if (command.equals("bye")) {
+            return 0;
+        } else if (command.equals("list")) {
+            taskList.list();
+            return 1;
+        } else if (command.startsWith("mark")) {
+            int item = Integer.parseInt(command.substring(command.length() - 1));
```

Comment:
What if this name was changed to `itemId` instead to help show that it's a number?